Emby.Server.Implementations/Channels/ChannelManager.cs

```diff
@@ -919,59 +961,55 @@ private BaseItem GetChannelItemEntity(ChannelItemInfo info, IChannel channelProv
+                        break;
+                    case ChannelFolderType.Series:
+                        item = GetItemById<Series>(info.Id, channelProvider.Name, out isNew);
+                        break;
+                    case ChannelFolderType.Season:
+                        item = GetItemById<Season>(info.Id, channelProvider.Name, out isNew);
+                        break;
+                    default:
+                        item = GetItemById<Folder>(info.Id, channelProvider.Name, out isNew);
+                        break;
                 }
             }
             else if (info.MediaType == ChannelMediaType.Audio)
             {
-                if (info.ContentType == ChannelMediaContentType.Podcast)
-                {
-                    item = GetItemById<AudioBook>(info.Id, channelProvider.Name, out isNew);
-                }
-                else
-                {
-                    item = GetItemById<Audio>(info.Id, channelProvider.Name, out isNew);
-                }
+                item = info.ContentType == ChannelMediaContentType.Podcast
+                    ? GetItemById<AudioBook>(info.Id, channelProvider.Name, out isNew)
+                    : GetItemById<Audio>(info.Id, channelProvider.Name, out isNew);
             }
             else
             {
-                if (info.ContentType == ChannelMediaContentType.Episode)
-                {
-                    item = GetItemById<Episode>(info.Id, channelProvider.Name, out isNew);
-                }
-                else if (info.ContentType == ChannelMediaContentType.Movie)
-                {
-                    item = GetItemById<Movie>(info.Id, channelProvider.Name, out isNew);
-                }
-                else if (info.ContentType == ChannelMediaContentType.Trailer || info.ExtraType == ExtraType.Trailer)
-                {
-                    item = GetItemById<Trailer>(info.Id, channelProvider.Name, out isNew);
-                }
-                else
+                switch (info.ContentType)
                 {
-                    item = GetItemById<Video>(info.Id, channelProvider.Name, out isNew);
+                    case ChannelMediaContentType.Episode:
+                        item = GetItemById<Episode>(info.Id, channelProvider.Name, out isNew);
+                        break;
+                    case ChannelMediaContentType.Movie:
+                        item = GetItemById<Movie>(info.Id, channelProvider.Name, out isNew);
+                        break;
+                    default:
+                        if (info.ContentType == ChannelMediaContentType.Trailer || info.ExtraType == ExtraType.Trailer)
+                        {
+                            item = GetItemById<Trailer>(info.Id, channelProvider.Name, out isNew);
+                        }
+                        else
+                        {
+                            item = GetItemById<Video>(info.Id, channelProvider.Name, out isNew);
+                        }
```

Comment:
This default block is kind of strange looking, I wonder if the original version was more readable.

I think you could also turn this into a switch expression. This should would work:

```cs
                item = info.ContentType switch
                {

                    ChannelMediaContentType.Episode => GetItemById<Episode>(info.Id, channelProvider.Name, out isNew),
                    ChannelMediaContentType.Movie => GetItemById<Movie>(info.Id, channelProvider.Name, out isNew),
                    var x when x == ChannelMediaContentType.Trailer || info.ExtraType == ExtraType.Trailer
                        => GetItemById<Trailer>(info.Id, channelProvider.Name, out isNew),
                    _ => GetItemById<Video>(info.Id, channelProvider.Name, out isNew)
                };
```